src/components/Todo.test.tsx

```diff
@@ -0,0 +1,65 @@
+import React from "react";
+import { fireEvent, render, screen } from "@testing-library/react";
+import Todo from "./Todo";
+import {
+  mockAddTodo,
+  mockFetchTodoList,
+  mockFetchTodoListError,
+} from "../mocks/mock";
+
+test("Has a submit button", () => {
+  render(<Todo fetchList={mockFetchTodoList} addTodo={mockAddTodo} />);
+
+  const submitButton = screen.getByTestId("add-button");
+
+  expect(submitButton).toBeInTheDocument();
+});
+
+test("Has an h1 header that says Todo List", () => {
+  render(<Todo fetchList={mockFetchTodoList} addTodo={mockAddTodo} />);
+
+  const header = screen.getByText("ToDo List:");
+
+  expect(header).toBeInTheDocument();
+});
+
+test("Task is added when button is pressed", () => {
+  render(<Todo fetchList={mockFetchTodoList} addTodo={mockAddTodo} />);
+
+  const input = screen.getByTestId("add-task");
+  fireEvent.change(input, { target: { value: "a new task" } });
+  const button = screen.getByRole("button");
+  fireEvent.click(button);
+  const li = screen.getByText("a new task");
+
+  expect(li).toBeInTheDocument();
+});
+
+test("Server is turned off", () => {
+  render(<Todo fetchList={mockFetchTodoListError} addTodo={mockAddTodo} />);
+  const button = screen.getByRole("button");
+
+  expect(button).toBeDisabled();
+});
+
+test("Error message pops up when server is turned off", () => {
+  render(<Todo fetchList={mockFetchTodoListError} addTodo={mockAddTodo} />);
+  const error = screen.getByTestId("error-message");
+
+  expect(error).toBeInTheDocument();
+});
+
+test("No error message pops up when server is turned on", () => {
+  render(<Todo fetchList={mockFetchTodoList} addTodo={mockAddTodo} />);
+  const error = screen.queryByTestId("error-message");
+
+  expect(error).not.toBeInTheDocument();
+});
+
+test("Error message is red with italics", () => {
+  render(<Todo fetchList={mockFetchTodoListError} addTodo={mockAddTodo} />);
+  const error = screen.queryByTestId("error-message");
```

Comment:
Why are you using `queryByTestId` instead of `getByTestId` here? Not a trick question here! Just curious.

Reply:
Using `queryByTestId` is the right approach for the previous test since you're trying to assert it isn't present. `getByTestId` was throwing an error because it couldn't find the node. So it makes sense to use `queryByTestId` instead. Here's a link to the [docs](https://testing-library.com/docs/queries/about/).

My next question is it best to use `queryByTestId` or `getByTestId` in this test? And why?

Reply:
I see what you're getting at. It would be best to use `getByTestId` if you were making sure something is in the document because it gives off errors if it can't find what you are looking for rather than `null` which I guess could give you false positives in some cases.